src/Streams.cpp

```diff
@@ -59,23 +52,29 @@ void Streams::check_stream_status() {
                 streams.end());
 }
 
-/**
- * Add a stream.
- *
- * @param s the stream to add.
- */
+/// Add a stream.
+///
+/// @param s the stream to add.
 void Streams::add(std::shared_ptr<Stream> s) { streams.push_back(s); }
 
-/**
- * Get the last stream in the vector.
- *
- * @return The last stream in the vector.
- */
+/// Get the last stream in the vector.
+///
+/// @return The last stream in the vector.
 std::shared_ptr<Stream> Streams::back() {
   return streams.empty() ? nullptr : streams.back();
 }
 
 const std::vector<std::shared_ptr<Stream>> &Streams::get_streams() {
   return streams;
 }
-}
\ No newline at end of file
+
+/// Return true is the given channel name is already forwarded
+bool Streams::hasChannelName(std::string const &channel_name) {
+  for (auto const &x : streams) {
+    if (x->channel_info().channel_name == channel_name) {
+      return true;
+    }
+  }
+  return false;
+}
+}
```

Comment:
Could this potentially use std::find_if?
Though this way is more readable... <shrug>